wkcuber/converter.py

```diff
@@ -354,18 +369,45 @@ def convert_input(self, args: Namespace) -> bool:
         ) = self.detect_dataset_name_and_layer_path_to_layer_name()
         put_default_if_not_present(args, "name", dataset_name)
 
+        bounding_box = None
+        view_configuration = dict()
         for layer_path, layer_name in layer_path_to_name.items():
-            args.layer_name = layer_name
             args.source_path = layer_path
-            cube_image_stack(
-                args.source_path,
-                args.target_path,
-                args.layer_name,
-                args.batch_size if "batch_size" in args else None,
-                args,
-            )
+            channel_count, dtype = get_channel_count_and_dtype(layer_path)
+            if channel_count > 1 and not (channel_count == 3 and dtype == "uint8"):
+                for i in range(channel_count):
+                    args.layer_name = layer_name + str(i)
+                    args.channel_index = i
+                    bounding_box = cube_image_stack(
+                        args.source_path,
+                        args.target_path,
+                        args.layer_name,
+                        args.batch_size if "batch_size" in args else None,
+                        args,
```

Comment:
can we unpack the other params, too, so that `args` hasn't to be passed additionally? at least, I'd pass `channel_index` separately to avoid mutation (then, parallelization over the channels also becomes easier).

Reply:
This comment is marked as resolved, but `args` is still passed as a whole? Were there any problems which blocked this change or did it slip through the cracks?

Reply:
I unpacked the `args`, but had to additionally pass them as a whole because of `with get_executor_for_args(executor_args) as executor:`. Additionally, the compression relies on the dtype detection of the cubing beforehand.